File: frontend/src/types/report.ts
export interface ReportArticle {
    article_id: number;
    title: string;
    authors: string[];
    journal?: string;
    publication_date?: string;
    pmid?: string;
    doi?: string;
    abstract?: string;
    url?: string;
    year?: string;
    // Association metadata
    relevance_score?: number;
    relevance_rationale?: string;
    ranking?: number;
    is_starred?: boolean;
    is_read?: boolean;
    notes?: string;
    presentation_categories?: string[];  // List of category IDs
}

export interface Report {
    report_id: number;
    user_id: number;
    research_stream_id: number | null;
    report_date: string;
    executive_summary?: string | null;  // Generated separately by LLM
    key_highlights: string[];
    thematic_analysis?: string | null;  // Generated separately by LLM
    coverage_stats: Record<string, any>;
    is_read: boolean;
    read_at: string | null;
    created_at: string;
    article_count?: number;
    // Pipeline execution metadata
    run_type?: string | null;  // 'test', 'scheduled', or 'manual'
    pipeline_metrics?: Record<string, any>;
    pipeline_execution_id?: string | null;  // UUID linking to WIP data
}

export interface ReportWithArticles extends Report {
    articles: ReportArticle[];
}

File: backend/schemas/report.py
"""
Report-related schemas for Knowledge Horizon
"""

from pydantic import BaseModel
from typing import List, Optional, Dict, Any
from datetime import datetime, date


class ReportArticle(BaseModel):
    """Article within a report with association metadata"""
    article_id: int
    title: str
    authors: List[str] = []
    journal: Optional[str] = None
    publication_date: Optional[str] = None
    pmid: Optional[str] = None
    doi: Optional[str] = None
    abstract: Optional[str] = None
    url: Optional[str] = None
    year: Optional[str] = None
    # Association metadata
    relevance_score: Optional[float] = None
    relevance_rationale: Optional[str] = None
    ranking: Optional[int] = None
    is_starred: Optional[bool] = False
    is_read: Optional[bool] = False
    notes: Optional[str] = None


class Report(BaseModel):
    """Report business object"""
    report_id: int
    user_id: int
    research_stream_id: Optional[int] = None
    report_date: date
    executive_summary: str
    key_highlights: List[str] = []
    thematic_analysis: str
    coverage_stats: Dict[str, Any] = {}
    is_read: bool = False
    read_at: Optional[datetime] = None
    created_at: datetime
    article_count: Optional[int] = None

    class Config:
        from_attributes = True


class ReportWithArticles(Report):
    """Report with full article details"""
    articles: List[ReportArticle] = []